python/pdstools/reports/ModelReport.qmd

```diff
@@ -139,21 +204,90 @@ The Cumulative Gains chart shows the percentage of he overall cases in the "posi
 
 The Lift chart is derived from this and shows the ratio of the cumulative gain and the targeted volume.
 
-TODO get both plots in
+TODO: unfortunately the Python version has no notion of unreachable bins (yet) which can cause really strange high lift values
+
+::: {layout-ncol=2}
+
+```{python}
+# TODO perhaps this should move into the pdstools plot functions "plotCumulativeGains"
+fig = px.area(
+    human_friendly_scoredistribution, 
+    x="Cum. Total (%)", y="Cum. Positives (%)", 
+    title='Cumulative Gains',
+    template="pega")
+fig.add_shape(
+    type='line', line=dict(dash='dash'),
+    x0=0, x1=100, y0=0, y1=100
+)
+fig.update_yaxes(scaleanchor="x", scaleratio=1, constrain='domain', title=dict(text="% of Positive Responders"))
+fig.update_xaxes(constrain='domain', title=dict(text="% of Population"))
+fig.show()
+```
+
+```{python}
+# TODO perhaps this should move into the pdstools plot functions "plotCumulativeLift"
+fig = px.area(
+    human_friendly_scoredistribution, 
+    x="Cum. Total (%)", y="Lift (%)", 
+    title='Lift',
+    template="pega")
+fig.update_yaxes(scaleanchor="x", scaleratio=0.01, constrain='domain', title=dict(text="Propensity Lift"))
+fig.update_xaxes(constrain='domain', title=dict(text="% of Population"))
+fig.show()
+```
+
+:::
 
 # Trend charts
 
-TODO using model datamart show trend of auc, responses, ctr, maybe all in
-one with tabs
+TODO see if we can put the various trend charts in tabs - although this seems to require go not express, then gain .data[0] probably gives the go objects
+
+But maybe .data[0] gives the trace
+
+
+```{python}
+fig = datamart.plotOverTime('weighted_performance')
+
+fig.update_layout(title="Model Performance Trend", yaxis_title="ROC-AUC", xaxis_title = "")
+fig.update_layout(showlegend=False)
+
+fig.show()
+```
+
+```{python}
+fig = datamart.plotOverTime('SuccessRate')
+
+fig.update_layout(title="Success Rate Trend", yaxis_title="Success Rate", xaxis_title = "")
+fig.update_layout(showlegend=False)
```

Comment:
Same as above - I'd personally put all `update_layout` arguments in one call, easier to change after